redash/query_runner/big_query.py

```diff
@@ -246,22 +250,106 @@ def _get_query_result(self, jobs, query):
+
+    def _columns_and_samples_to_dict(self, schema, samples):
+        samples_dict = {}
+        if not samples:
+            return samples_dict
+
+        # If a sample exists, its shape/length should be analogous to
+        # the schema provided (i.e their lengths should match up)
+        for i, column in enumerate(schema):
+            if column['type'] == 'RECORD':
+                associated_sample = samples[i]
+                if column.get('mode', None) == 'REPEATED':
+                    # Repeated fields have multiple samples of the same format.
+                    # We only need to show the first one as an example.
+                    associated_sample = [] if len(samples[i]) == 0 else samples[i][0]
+
+                for j, field in enumerate(column['fields']):
+                    col_name = u"{}.{}".format(column['name'], field['name'])
+                    samples_dict[col_name] = None
+                    if type(associated_sample) == list and len(associated_sample) > 0:
+                        samples_dict[col_name] = associated_sample[j]
+            else:
+                samples_dict[column['name']] = samples[i]
+
+        return samples_dict
+
+
+    def _flatten_samples(self, samples):
+        samples_list = []
+        for field in samples:
+            value = field['v']
+            if type(value) == dict:
+                samples_list.append(
+                    self._flatten_samples(value.get('f', []))
+                )
+            elif type(value) == list:
+                samples_list.append(
+                    self._flatten_samples(value)
+                )
+            else:
+                samples_list.append(value)
+
+        return samples_list
+
+    def get_table_sample(self, table_name):
+        if not self.configuration.get('loadSchema', False):
+            return {}
+
+        service = self._get_bigquery_service()
+        project_id = self._get_project_id()
+
+        dataset_id, table_id = table_name.split('.')
 
-        return columns
+        try:
+            sample_response = service.tabledata().list(
+                projectId=project_id,
+                datasetId=dataset_id,
+                tableId=table_id,
+                maxResults=1
+            ).execute()
+            schema_response = service.tables().get(
+                projectId=project_id,
+                datasetId=dataset_id,
+                tableId=table_id,
+            ).execute()
+            table_rows = sample_response.get('rows', [])
+            samples = ({} if len(table_rows) == 0 else table_rows[0]).get('f', [])
+            schema = schema_response.get('schema', {}).get('fields', [])
+            columns = self._get_columns_schema(schema_response).get('columns', [])
+
+            flattened_samples = self._flatten_samples(samples)
+            samples_dict = self._columns_and_samples_to_dict(schema, flattened_samples)
+            return samples_dict
+        except Exception as e:
```

Comment:
Let's be more specific what exceptions could happen here to not accidentally hiding things we aren't expecting. The `logger.exception` call right now doesn't do much right now since we don't look for it.